uniffi_core/src/ffi/rustbuffer.rs

```diff
@@ -271,6 +274,7 @@ pub fn uniffi_rustbuffer_reserve(
             .expect("additional buffer length negative or overflowed");
         let mut v = buf.destroy_into_vec();
         v.reserve(additional);
+        v.shrink_to(i32::MAX as usize);
         Ok(RustBuffer::from_vec(v))
```

Comment:
The same effect could be achieved more efficiently using `v.reserve(std::cmp::min(additional, i32::MAX - v.len()));`.

Reply:
My understanding is that [reserve](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve) will reserve at least `additional` more elements, but it _could_ reserve more, which would put it in danger of exceeding the max capacity. `shrink_to` is a no-op if the current capacity is less than the requested minimum, so I think the cost of `shrink_to` is fairly light for `Vec`s that wouldn't exceed the capacity anyway.

Reply:
How about `reserve_exact`?